test/cmocka/src/math/arithmetic/exponential.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include <sof/math/exp_fcn.h>
+#include <sof/audio/format.h>
+#include <rtos/string.h>
+#include <sof/common.h>
+
+#define ULP_TOLERANCE 5.60032793
+#define ULP_SCALE 0.0000999999999749
+
+static void gen_exp_testvector(double a, double b, double *r, int32_t *b_i);
+
+/* testvector in Q4.28, -5 to +5 */
+/*
+ * Arguments	:double a
+ *		 double b
+ *		 double *r
+ *		 int32_t *b_i
+ * Return Type	:void
+ */
+static void gen_exp_testvector(double a, double b, double *r, int32_t *b_i)
+{
+	double a2;
+	double b2;
+
+	a2 = a / 2.0;
+	b2 = b / 2.0;
+
+	*r = a + rand() % (int32_t)(b - a + 1);
+	a2 = *r * 268435456;
+	b2 = fabs(a2);
+	if (b2 < 4503599627370496)
+		a2 = (b2 < 4503599627370496) ? floor(a2 + 0.5) : 0.0;
```

Comment:
As reported when you ran `ccpcheck`, why are you checking this again? You know it's true because it's just been checked on the previous line.
```suggestion
		a2 = floor(a2 + 0.5);
```

I realize this is only test code but this generally lowers the confidence in the rest of the code.

Reply:
done